tests/neptune/test_artifacts.py

```diff
@@ -206,3 +207,36 @@ def test_local_existing(self, run: Run):
 
         assert run[first].fetch_hash() == run[second].fetch_hash()
         assert run[first].fetch_files_list() == run[second].fetch_files_list()
+
+    def test_hash_cache(self, run: Run):
+        first, second, third = self.gen_key(), self.gen_key(), self.gen_key()
+        filename = fake.file_name()
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with preserve_cwd(tmp):
+                # create 2GB file
+                with open(filename, 'wb') as handler:
+                    handler.write(b'\0' * 2 * 2 ** 30)
+
+                # track it
+                start = time.time()
+                run[first].track_files('.', wait=True)
+                initial_duration = time.time() - start
+
+                # and track it again
+                start = time.time()
+                run[first].track_files('.', wait=True)
```

Comment:
`run[second]`

Reply:
I've just updated it by logging everything to one key, will it be a problem since we do keep duplicateds?

Reply:
That's fine.